user-service/README.md

```diff
@@ -1,26 +1,104 @@
+The User Service separates this concern from the rest of the system. Other components use this service's API to request user details. 
+Some deployments may also allow Palisade to add users to the system, hence the presence of the `addUser()` method in the `UserService` interface.
 
-## Overview
-The User service is responsible for providing the other Palisade components with knowledge of the users of the system. Note that we are avoiding having a separate notion of "Palisade"" users that are distinct from the normal user accounts of the system. There is no such thing as a "Palisade"" user.
+## Message Model and Database Domain
 
-One of the purposes of this service is to allow Palisade to adopt the whatever notion of *user* the host environment has. For example, this may from a central directory service such as LDAP, host operating system account or PKI based user authentication.
+| UserRequest     | UserResponse     | AuditErrorMessage | AuditableUserResponse |
+|:----------------|:-----------------|:------------------|:----------------------|
+| *token          | *token           | *token            | *token                | 
+| userId          | userId           | userId            | UserResponse          |  
+| resourceId      | resourceId       | resourceId        | AuditErrorMessage     |
+| context         | context          | context           |                       |
+|                 | User             | exception         |                       | 
+|                 |                  | serverMetadata    |                       |
+  
+(fields marked with * are acquired from headers metadata)
 
-The user service separates this concern from the rest of the system. Other components use this service's API to request user details. Some deployments may also allow Palisade to add users to the system, hence the presence of the `addUser()` method in the `UserService` interface.
+The service accepts a `UserRequest` and a token, then checks that the User is in the cache; the technology of which is chosen in the User Services application.yaml and Spring profiles.
+If the User exists in the cache, then it, alongside any attributes are retrieved, and used to create a `UserReponse` object which is then packaged into a `AuditableUserResponse` object with no errors, 
+and then this is sent via kafka, to the Resource Service for further processing. 
+If the User does not exist in the cache, then a `NoSuchUserIdException` is thrown within the Service, which is caught, and packaged in the `AuditableUserReponse`, alongside the original `UserRequest` and an `AuditErrorMessage`, 
+this is then sent to the Audit Service and logged appropriatly.
 
-## API Usage
+## REST Interface
 
-Currently the `UserService` supports adding and retrieving of user details via the two following API calls.
+The application exposes two REST endpoints used for debugging or mocking kafka entrypoints:
+* `POST api/user`
```

Comment:
This would require a change to the other services also, as they follow the same convention of /api/*service e.g /api/mask
if its worth doing, we can create a ticket on the backlog

Reply:
I think it is worth doing, but is a low priority. An issue in the backlog would be best I think.

Reply:
I like the usage we currently have as it is meant for pushing items onto kafka queues. In this way, the REST endpoint reflects the name of the topic.